src/publish.proj

```diff
@@ -1,41 +1,66 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
-  <Import Project="$(ToolsDir)PublishContent.targets" />
-  <Import Project="$(ToolsDir)versioning.targets" />
 
-  <!-- gathers the items to be published -->
-  <Target Name="GatherItemsForPattern">
-    <Error Condition="'$(PublishPattern)' == ''" Text="Please specify a value for PublishPattern using standard msbuild 'include' syntax." />
-    <ItemGroup>
-      <ForPublishing Include="$(PublishPattern)" />
-    </ItemGroup>
-    <!-- add relative blob path metadata -->
-    <ItemGroup>
-      <ForPublishing>
-        <RelativeBlobPath Condition="'$(PublishTestNativeBins)' != 'true'">$(__BuildType)/%(RecursiveDir)%(Filename)%(Extension)</RelativeBlobPath>
-        <RelativeBlobPath Condition="'$(PublishTestNativeBins)' == 'true'">$(__BuildType)/%(RecursiveDir)%(Filename)%(Extension)</RelativeBlobPath>    
-      </ForPublishing>
-    </ItemGroup>
-    <Error Condition="'@(ForPublishing)' == ''" Text="No items were found matching pattern '$(PublishPattern)'." />
-  </Target>
+  <PropertyGroup>
+    <FeedsDir Condition="'$(BuildOS)' == 'Windows_NT'">$(ToolsDir)Microsoft.Dotnet.Build.Tasks.Feed/desktop/</FeedsDir>
+    <FeedsDir Condition="'$(BuildOS)' != 'Windows_NT'">$(ToolsDir)Microsoft.Dotnet.Build.Tasks.Feed/</FeedsDir>
+  </PropertyGroup>
+
+  <UsingTask TaskName="PushToBlobFeed" AssemblyFile="$(FeedsDir)Microsoft.DotNet.Build.Tasks.Feed.dll"/>
+
+  <PropertyGroup>
+    <PackagesPattern Condition="'$(PackagesPattern)' == '' and '$(PublishTestNativeBins)' != 'true'">$(PackagesBinDir)pkg\*.nupkg</PackagesPattern>
+    <PackagesPattern Condition="'$(PackagesPattern)' == '' and '$(PublishTestNativeBins)' == 'true'">$(OutputPath)\bin\**</PackagesPattern>
+    <SymbolsPackagesPattern>$(PackagesBinDir)symbolpkg\*.nupkg</SymbolsPackagesPattern>
+  </PropertyGroup>
 
   <PropertyGroup>
-    <PublishPattern Condition="'$(PublishPattern)' == '' and '$(PublishTestNativeBins)' != 'true'">$(PackagesBinDir)**\*.nupkg</PublishPattern>
-    <PublishPattern Condition="'$(PublishPattern)' == '' and '$(PublishTestNativeBins)' == 'true'">$(OutputPath)\bin\**</PublishPattern>
+    <ContainerName Condition="'$(__Container)' == '' and '$(PublishTestNativeBins)' != 'true'">coreclr-$(PreReleaseLabel)-$(BuildNumberMajor)-$(BuildNumberMinor)</ContainerName>
+    <ContainerName Condition="'$(__Container)' == '' and '$(PublishTestNativeBins)' == 'true'">coreclr-$(PreReleaseLabel)-$(BuildNumberMajor)-$(BuildNumberMinor)-$(__DistroRid)-$(__BuildArch)</ContainerName>
+    <ContainerName Condition="'$(__Container)' != '' and '$(PublishTestNativeBins)' != 'true'">$(__Container)</ContainerName>
```

Comment:
Where would you expect __Containter to ever be set?

Reply:
It's set here - https://github.com/dotnet/coreclr/blob/master/config.json#L598-L601

Reply:
When the user calls publish-packages.cmd -Container=xxx

Reply:
I would suggest we change that to just change the ContainerName property and eliminate this extra logic. In general properties starting with __ are supposed to be private and only used locally so I would prefer we not depend on an extra one here.

Reply:
Given that the container keeps changing, do you know how you are going to track restoring?

Reply:
I wasn't sure if that was covered by the new restore mechanism. Should I always publish into the same container, but with a different RelativeBlobPath? Is there a defined scheme for the container name I should use?

Reply:
Yes use the same container with a different RelativeBlobPath. The proposal for a scheme is at https://github.com/dotnet/core-eng/issues/753.